### PR TITLE
Compute `tf.gather`'s output shape instead of passing the table through

### PR DESCRIPTION
**Pull request by khatchad**
Fixes wala/ML#815.

## The Cause

`tf.gather` was modeled as `pass_through`, so it returned its `params` argument's type. A subject reading

```python
embeddings = K.gather(self.embeddings, inputs)
```

therefore typed the result as the embedding table itself, `(vocab, model_dim)`, rather than the lookup, `(batch, sequence, model_dim)`. That explains the "table-like extents" the issue reports: `(5000, 8)` is the table, and `5000` is the vocabulary size.

Three wirings carried the pass-through: `tf.gather`, `array_ops.gather`, and the Keras backend alias, which is the form the subject uses.

## Why It Matters More Than Imprecision

An argument of a different rank is not a subtype of the specification, so a signature written from this rejects every call the function receives, in the same way as wala/ML#813 though by a different route.

## The Fix Reuses An Existing Rule

`tf.nn.embedding_lookup` already computes `ids.shape + params.shape[1:]`, because an embedding lookup **is** a gather along the first axis. `Gather` therefore subclasses `EmbeddingLookup` rather than restating the arithmetic in a second place that can drift from the first.

Scoped to the default first-axis gather. A non-default `axis` or a non-zero `batch_dims` rearranges the result differently and is left to the inherited behavior rather than answered wrongly, which the class says explicitly.

## A Note On The Model

Rewiring the field alone was not enough: with `<new>` and the `putfield` but no synthetic class body, the call produced nothing and the parameter stopped being tensor-typed at all. The `do()` method with `<new>` and `<return>` is what makes an allocating model yield a tensor, following `gather_nd` exactly.

## Testing

Two fixtures, both run to completion under `python3.10` with their assertions, which the JUnit expectations mirror: the direct call and the Keras backend alias, since real code commonly reaches it through the latter.

Full module suite: 1195 tests, 0 failures, 3 skipped.

## Provenance

Found by comparing emitted signatures against argument types observed at run time, and reproduced through a second driver before being filed.

### CODE REVIEW

**Comment by khatchad** [comment]
Correction to this comment: `Gather` did not need a second registration, and the manual arm I added here was dead code. I have reverted it.

`TensorGeneratorFactory.dispatchShared` is consulted first by both `getGeneratorBody` and `createManualGenerator`, and the `GATHER` arm there already names both construction forms through `anchor.makeGenerator(Gather::new, Gather::new)`. That is the shared chain wala/ML#760 introduced precisely so an operation cannot be registered for one anchoring and not the other. Registering in it once is the whole fix; the `else if` I added below it was unreachable, which is what the patch-coverage report was reporting.

What remains from this branch is a test that consumes the gathered value as an elementwise operand rather than only as a sink argument.

Full module suite: 1195 tests, 0 failures, 3 skipped.
